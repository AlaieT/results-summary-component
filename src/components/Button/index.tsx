import React from "react";

import type { ButtonProps } from "../../types";

import styles from "../../styles/components/button.module.scss";

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  ({ children, type, ...rest }, ref) => (
    <button {...rest} ref={ref} type={type} className={styles.button}>
      {children}
    </button>
  )
);

export default Button;
