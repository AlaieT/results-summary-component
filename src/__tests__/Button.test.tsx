import React from "react";
import { render, cleanup, screen, fireEvent } from "@testing-library/react";

import Button from "../components/Button";

afterEach(cleanup);

describe("Button", () => {
  const mockFunc = jest.fn((e) => e.preventDefault());

  describe("correct render", () => {
    it("should match snapshot", () => {
      const { asFragment } = render(<Button />);

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("correct functionality", () => {
    it("should call function on click if type - button", async () => {
      render(
        <Button type="button" onClick={mockFunc}>
          Click
        </Button>
      );

      fireEvent.click(await screen.findByText("Click"));

      expect(mockFunc).toBeCalled();
    });

    it("should submit form on click if type - submit", async () => {
      render(
        <form onSubmit={mockFunc}>
          <Button type="submit">Submit</Button>
        </form>
      );

      fireEvent.click(await screen.findByText("Submit"));

      expect(mockFunc).toBeCalled();
    });
  });
});
