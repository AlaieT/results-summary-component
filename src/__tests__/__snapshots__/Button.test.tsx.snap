// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button correct render should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
  />
</DocumentFragment>
`;
